api/v1alpha1/dataprotectionapplication_types.go

```diff
@@ -848,6 +862,23 @@ func (dpa *DataProtectionApplication) AutoCorrect() {
 
 	dpa.Spec.Configuration.Velero.DefaultPlugins = common.RemoveDuplicateValues(dpa.Spec.Configuration.Velero.DefaultPlugins)
 	dpa.Spec.Configuration.Velero.FeatureFlags = common.RemoveDuplicateValues(dpa.Spec.Configuration.Velero.FeatureFlags)
+
+	// Auto correct nodeAffinity for the node agent, but only if the new schema is not used
+	// The new schema will be used instead of the dpa.Spec.Configuration.NodeAgent.PodConfig
+	// There is need to translate map of labels to a LabelSelector with MatchLabels
+	if dpa.Spec.Configuration.NodeAgent != nil && dpa.Spec.Configuration.NodeAgent.PodConfig != nil && dpa.Spec.Configuration.NodeAgent.PodConfig.NodeSelector != nil {
```

Comment:
I was worried about this

example: 
> I create DPA using old style (NodeAgent.PodConfig.NodeSelector), code will make me use new style NodeAgent.LoadAffinityConfig
>
> But once I update DPA NodeAgent.PodConfig.NodeSelector, NodeAgent.LoadAffinityConfig will get out of date, right? and then node agent will not have node selector user wants. Can this create problems?

but the validator controller errors if both objects are not equal, right?

Reply:
but if user update is to add a new label to old field, then ConfigMap will not reflect what it is expecting, right? like the test I got worried abot

Reply:
No, the test involves combining both old and new configurations.

In this path we're only focusing on PodConfig.

Could you please provide the steps to reproduce the potential issue you're thinking about, so I can better understand what you're referring to, as I might have overlooked something.

Reply:
Could not reproduce problem in cluster

Reply:
I was testing this scenario with updating and for me was ok, we are using patch of configmap. We should be good here, right?